RandLAPACK/drivers/rl_cqrrpt.hh

```diff
@@ -112,318 +111,218 @@ class CQRRPT : public CQRRPTalg<T, RNG> {
-    }
 
     // Swap k columns of A with pivots from J
-    util::col_swap(m, n, k, A_dat, m, J);
+    blas::copy(n, J, 1, J_buf.data(), 1);
+    util::col_swap(m, n, k, A, lda, J_buf);
 
     if(this -> timing) {
         a_mod_piv_t_stop = high_resolution_clock::now();
         a_mod_trsm_t_start = high_resolution_clock::now();
     }
 
     // A_pre * R_sp = AP
-    blas::trsm(Layout::ColMajor, Side::Right, Uplo::Upper, Op::NoTrans, Diag::NonUnit, m, k, 1.0, R_sp_dat, k, A_dat, m);
+    blas::trsm(Layout::ColMajor, Side::Right, Uplo::Upper, Op::NoTrans, Diag::NonUnit, m, k, 1.0, R_sp, k, A, lda);
 
-    if(this -> timing)
+    if(this -> timing) {
         a_mod_trsm_t_stop = high_resolution_clock::now();
-
-    // Check the condition number of a A_pre
-    if(this -> cond_check)
-    {
-        // Check cond(A_pre)
-        std::vector<T> A_pre_cpy;
-        std::vector<T> s;
-        this->cond_num_A_pre = RandLAPACK::util::cond_num_check(m, k, A, A_pre_cpy, s, false);
-
-        A_pre_cpy.clear();
-        s.clear();
-        // Check cond(normc(A_pre))
-        std::vector<T> A_norm_pre;
-        RandLAPACK::util::normc(m, k, A, A_norm_pre);
-        this->cond_num_A_norm_pre = RandLAPACK::util::cond_num_check(m, k, A_norm_pre, A_pre_cpy, s, false);
-    }
-
-    if(this -> timing)
         cholqr_t_start = high_resolution_clock::now();
+    }
 
     // Do Cholesky QR
-    blas::syrk(Layout::ColMajor, Uplo::Upper, Op::Trans, k, m, 1.0, A_dat, m, 0.0, R_sp_dat, k);
-    if(lapack::potrf(Uplo::Upper, k, R_sp_dat, k)){
-        if(this->verbosity)
-            throw std::runtime_error("Cholesky decomposition failed.");
-        return 1;
-    }
+    blas::syrk(Layout::ColMajor, Uplo::Upper, Op::Trans, k, m, 1.0, A, lda, 0.0, R_sp, k);
+    lapack::potrf(Uplo::Upper, k, R_sp, k);
 
     // Re-estimate rank after we have the R-factor form Cholesky QR.
     // The strategy here is the same as in naive rank estimation.
     // This also automatically takes care of any potentical failures in Cholesky factorization.
-    // Note that the diagonal of R_sp_dat may not be sorted, so we need to keep the running max/min
-    // We expect the loss in the orthogonality of Q to be approximately equal to u * cond(R_sp_dat)^2, where u is the unit roundoff for the numerical type T.
-    int64_t new_rank = k;
-    T running_max = R_sp_dat[0];
-    T running_min = R_sp_dat[0];
-    T curr_entry;
+    // Note that the diagonal of R_sp may not be sorted, so we need to keep the running max/min
+    // We expect the loss in the orthogonality of Q to be approximately equal to u * cond(R_sp)^2, where u is the unit roundoff for the numerical type T.
+    new_rank = k;
+    running_max = R_sp[0];
+    running_min = R_sp[0];
     
     for(i = 0; i < k; ++i) {
         curr_entry = std::abs(R_sp[i * k + i]);
-        
         if(curr_entry > running_max) running_max = curr_entry;
         if(curr_entry < running_min) running_max = running_min;
```

Comment:
@TeachRaccooon this never updates ``running_min``. I think line 288 should be ``running_min = curr_entry``.

Reply:
Fixed.